File: MDP.m
%"Markov Decision Process for Battery usage optimization"
%States, where bigger is more charged
states = [1,2,3,4];
actions = ["navigate1","navigate2","idle","recharge"]; 
rewards =  [[10,6,4,-10],...
            [7,5,4,-5],...
            [2,4,5,1],...
            [-15,-10,-7,10]]; %Maybe? idk - YE
discount = 0.85; 
load("trans_mats")
disp(matrices)

File: MDP.py
import numpy as np

class MDP:
    def __init__(self,s,a,tp,r,df):
        """
        s - States of MDP
        a - Actions the agent can take
        tp - transition probabilities
        r - rewards
        df - discount factor
        """
        self.s = s
        self.a = a
        self.tp = tp
        self.r = r
        self.df = df
    
